### PR TITLE
Added new page for Routing and Obstacles in the System

### PR DESCRIPTION
**Pull request by AngelicaRoss**

### CODE REVIEW

File: docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+sidebar_position: 1
+---
+
+# Routing and Obstacles in the System
+
+This page provides an overview of routing and the different obstacles delaying or preventing routes from being found. It can act as a useful checklist for when a task is not being performed.
+
+## Routing🛣️
+
+Routing is the process of find the path an Omni-Mole will take to perform a task. It is created by analysing the current state of the system, specifically the zones the Omni-Mole is in, and choosing the best path that navigates around any obstacles in the system.
```
Comment (itstudes) [inline docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md:11]: Perhaps rephrase this sentence "Routing is the process of find the path.." hmmm

File: docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+Different tasks require different routes, this is discussed below.
+
+#### Generic Tasks🛣️
+
+Generic Tasks require a single route from the Omni-Moles current location to the tasks destination location.
+
+#### Pallet Task🚚
+
+Pallet Tasks require a route from the Omni-Moles current location to the pallets current location and then another route from the pallets current location to the tasks destination location. If the ssecond route cannot be performed then the first route will also not be performed.
```
Comment (itstudes) [inline docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md:31]: I think you need some apostrophes here. "Pallet Tasks require a route from the Omni-Mole's current location to the pallet's current location.." Etc.

Comment (itstudes) [reply docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md:31]: Actually, apostrophes needed throughout the file

File: docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+#### Multi-Pallet Task🔁
+
+Multi-Pallet Tasks require a single route from the Omni-Moles current location to the tasks destination location.
+
+## Obstacles🚧
+
+Obstacles are defines as any obstruction to the routing of the system, such as locks, Omni-Moles, or zone setup. It is important to understand these when diagnosing issues related to routing.
```
Comment (itstudes) [inline docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md:39]: defined

File: docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+### Temporary Obstacles🚦
+
+Temporary obstacles are obstacles that will automatically be resolved by the system and so a path can be found by passing through them. The Omni-Mole will move along it's path until it reaches a temporary obstacle, wait for it to be resolved, then continue. A couple examples of temporary obstacles are active subtasks, Omni-Moles assigned to tasks and Omni-Moles the are online and in auto and can be relocated.
```
Comment (itstudes) [inline docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md:43]: ".. and Omni-Moles THAT are online .."

File: docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+### Permanent Obstacles⛔
+
+Permanent obstacles can not be automatically resolved and paths will need to be routed around them. If no path can be found by routing around them, then the task cannot continue until a user manually resolves the obstacles blocking the path.
```
Comment (itstudes) [inline docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md:47]: cannot

File: docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+Below is an example of a manual lock blocking an Omni-Mole from reaching a pallet.
+![Toolbar Location](assets/routing-and-obstacles-in-the-system/permanent-obstacle-lock-blocking-source.png)
+
+##### Solution💡
+
+In order to resolve the issue the locks need to be removed.
+
+###### Manual Locks🚫
+
+Manual locks can only be create by a user so it is best to check why this lock was created before removing it.
```
Comment (itstudes) [inline docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md:67]: created

File: docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+Below is an example of an offline Omni-Mole preventing a path from being found.
+![Toolbar Location](assets/routing-and-obstacles-in-the-system/permanent-obstacle-omnimole-blocking-source.png)
+
+##### Solution💡
+
+Check why the Omni-Mole is offline and then either put it back online or manually move it out of the way and update it's location.
+
+#### Omni-Mole Not Assigned the Zone
+
+If an Omni-Mole is not assigned to a zone that include the source and destination or a path between them then a path cannot be found.
```
Comment (itstudes) [inline docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md:109]: I think this sentence needs some commas!

Comment (itstudes) [reply docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md:109]: Just check elsewhere for commas, you can break up some of these sentences a bit.

**Review by itstudes — APPROVED** [review]
This looks great, you're welcome to merge once you've fixed those minor grammar issues

File: docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md (hunk 7 above)
Comment (AngelicaRoss) [reply docs/04-Basic-Functionality/02-Managing-the-System/Routing and Obstacles in the System.md:109]: I became super selfconcious about comman splicing🥲 but will go through it again and see where I missed them

**Pull request merged**